flink-connectors/flink-connector-jdbc/src/test/java/org/apache/flink/connector/jdbc/table/UnsignedTypeConversionITCase.java

```diff
@@ -60,27 +60,54 @@
 	private static final String TABLE_NAME = "unsigned_test";
 
 	private StreamTableEnvironment tEnv;
+	private DB db;
 	private String dbUrl;
 	private Connection connection;
 
-	@ClassRule
-	public static MariaDB4jRule db4jRule = new MariaDB4jRule(
-		DBConfigurationBuilder.newBuilder().build(),
-		DEFAULT_DB_NAME,
-		null);
-
 	@Before
-	public void setUp() throws SQLException, ClassNotFoundException {
-		StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
-		tEnv = StreamTableEnvironment.create(env);
+	public void setUp() throws SQLException, IllegalStateException {
 		//dbUrl: jdbc:mysql://localhost:3306/test
-		dbUrl = db4jRule.getURL();
-		connection = DriverManager.getConnection(dbUrl);
+		prepareMariaDB();
 		createMysqlTable();
+		StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+		tEnv = StreamTableEnvironment.create(env);
 		createFlinkTable();
 		prepareData();
 	}
 
+	private void prepareMariaDB() throws IllegalStateException {
+		boolean initDbSuccess = false;
+		int i = 0;
+		//retry
```

Comment:
Please add more comments for why we add a retry here, we can also link the issue.